RockLib.Configuration.Remote/JsonConfigurationParser.cs

```diff
@@ -0,0 +1,77 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace RockLib.Configuration.Remote;
+
+/// <summary>
+/// A parse for JSON configuration.
+/// </summary>
+public class JsonConfigurationParser : IConfigurationParser
+{
+    private const string SectionSeparator = ":";
+
+    private readonly string _section;
+
+    /// <summary>
+    /// Create a JsonConfigurationParser instance with keys rooted at a given
+    /// section path.
+    /// </summary>
+    /// <param name="section">The section path to append parsed configuration into</param>
+    public JsonConfigurationParser(string section)
+    {
+        _section = section;
+    }
+
+    /// <summary>
+    /// Parse configuration from a raw JSON string.
+    /// </summary>
+    /// <param name="raw">The raw JSON string</param>
+    /// <returns>A Dictionary of configuration, rooted at the given section</returns>
+    public IDictionary<string, string> Parse(string raw)
```

Comment:
This limits the structure of the remote configuration to be basic key value pairs which may be fine initially, however, we will need to have documentation that talks about the format of the remote configuration.

Reply:
This is exactly how .NET configuration works, actually. Configuration is always flattened, but it can be mapped into a non-flat structure using the path delimiters.

This does come with a limitation where multiple configuration sources cannot _append_ values into the same array, but that limitation has always been there and is not introduced by this implementation.

Reply:
An example of that limitation: If we had these two JSON files:

### appsettings.json

```json
{
  "Items": [1, 2, 3]
}
```

### appsettings.Development.json

```json
{
  "Items": [4]
}
```

and we mapped `Configuration.GetSection("Items")` into a `List<int>`, we'd get `[4, 2, 3]`.